RedHatInsights/rhsm-subscriptions
# SWATCH-1188: Remove metering JMX endpoint and replace with REST endpoint

See https://issues.redhat.com/browse/SWATCH-1188

# Testing

1.
   ```bash
   ./gradlew :bootRun --args="--spring.profiles.active=kafka-queue,openshift-metering-worker --rhsm-subscriptions.enableSynchronousOperations=true --rhsm-subscriptions.auth.swatchPsks.self=secret"
   ```
1. Run metering for a specific org.
   ```bash
   http POST :8000/api/rhsm-subscriptions/v1/internal/metering/OpenShift-metrics orgId==org123 rangeInMinutes==60 x-rh-swatch-synchronous-request:true x-rh-swatch-psk:secret
   ```
1. You will see log messages indicating the metering has started.  However, unless you are connected to a Prometheus instance, you'll see "connection refused" messages.
1. Run metering for all orgs.
   ```bash
   http POST :8000/api/rhsm-subscriptions/v1/internal/metering/OpenShift-metrics rangeInMinutes==60 x-rh-swatch-synchronous-request:true x-rh-swatch-psk:secret
   ```
   This will likewise fail, but you'll see a message reading "Metering OpenShift-metrics for all accounts in the past 60 minutes" above the "Connection Refused" stacktrace.
